### PR TITLE
Job Detail Modal

### PR DESCRIPTION
**Pull request by jaredwebber**
For #32 
-> Created #53 to connect backend once complete

## Summary Of Changes
Created Modal for editing application info.
Clicking a kanban card opens modal populated with job info, and clicking 'Add New Application' Button opens blank modal, and adds the card to Kanban when `save` is clicked.

Screenshot 2023-02-08 at 1 27 09 PM

## Notes for Reviewers
- Only fields which are required to be non-empty are `position`, `company` and `status` (column)

### CODE REVIEW

**Review by drkitt — CHANGES_REQUESTED** [review]
The code looks good but I found a couple of functionality issues:
- The Add Deadline button gets cut off on a small browser window 
Screenshot 2023-02-09 at 11 58 17 AM
- It's possible to save a deadline without a name or date

**Review by drkitt — APPROVED** [review]
All of my issues resolved, LGTM

File: frontend/src/components/modal/job/JobDetailModal.vue [hunk 1]
```diff
@@ -0,0 +1,316 @@
+<template>
+  <v-row justify="center">
+    <v-dialog v-model="dialog" persistent>
+      <v-card>
+        <v-card-text>
+          <v-row>
+            <v-col cols="" sm="" class="info-col">
```
Comment (Speuce) [inline frontend/src/components/modal/job/JobDetailModal.vue:7]:
```suggestion
            <v-col class="info-col">
```

File: frontend/src/components/modal/job/JobDetailModal.vue [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+              <v-row>
+                <v-col cols="12" sm="6">
+                  <v-text-field
+                    label="Position*"
+                    class="text-h5"
+                    v-model="jobData.position"
+                    :style="{ color: this.positionErrorIndicator }"
```
Comment (Speuce) [inline frontend/src/components/modal/job/JobDetailModal.vue:14]: Should this input be restricted to numeric? I think there's a `number` prop you can pass in.

File: frontend/src/components/modal/job/JobDetailModal.vue [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+                  ></v-text-field
+                ></v-col>
+
+                <v-col cols="12" sm="1"></v-col>
+
+                <v-col cols="12" sm="5">
+                  <v-text-field
+                    label="Type"
+                    v-model="jobData.type"
+                  ></v-text-field>
```
Comment (Speuce) [inline frontend/src/components/modal/job/JobDetailModal.vue:24]: Can we enforce some sort of validation here? Maybe max 255 chars or something.

File: frontend/src/components/modal/job/JobDetailModal.vue [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+              </v-row>
+
+              <v-row
+                ><v-col cols="12" sm="6">
+                  <v-text-field
+                    label="Company*"
+                    required
+                    v-model="jobData.company"
+                    :style="{ color: this.companyErrorIndicator }"
+                  ></v-text-field>
```
Comment (Speuce) [inline frontend/src/components/modal/job/JobDetailModal.vue:35]: Can we enforce some sort of validation here? Maybe max 255 chars or something.

File: frontend/src/components/modal/job/JobDetailModal.vue [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+
+              <v-row>
+                <v-col cols="12" sm="">
+                  <v-textarea
+                    auto-grow
+                    class="text-area-box"
+                    label="Description"
+                    shaped
+                    v-model="jobData.description"
+                  ></v-textarea
```
Comment (Speuce) [inline frontend/src/components/modal/job/JobDetailModal.vue:59]: Might also want some validation here, like 1-10k chars or something.

File: frontend/src/components/modal/job/JobDetailModal.vue [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+              </v-row>
+
+              <v-row>
+                <v-col cols="12" sm="">
+                  <v-textarea
+                    auto-grow
+                    class="text-area-box"
+                    label="Cover Letter"
+                    shaped
+                    v-model="jobData.coverLetter"
```
Comment (Speuce) [inline frontend/src/components/modal/job/JobDetailModal.vue:70]: Might also want some validation here, like 1-10k chars or something.

File: frontend/src/components/modal/job/JobDetailModal.vue [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+          >
+            Ensure Required (*) Fields Are Filled
+          </h4>
+        </v-card-text>
+        <v-card-actions>
+          <v-spacer></v-spacer>
+          <v-btn
+            color="blue-darken-1"
+            variant="text"
+            @click="this.closeClicked()"
```
Comment (Speuce) [inline frontend/src/components/modal/job/JobDetailModal.vue:129]:
```suggestion
            @click="this.closeClicked"
```

File: frontend/src/components/modal/job/JobDetailModal.vue [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+          >
+            Close
+          </v-btn>
+          <v-btn
+            color="blue-darken-1"
+            variant="text"
+            @click="this.saveClicked()"
```
Comment (Speuce) [inline frontend/src/components/modal/job/JobDetailModal.vue:136]:
```suggestion
            @click="this.saveClicked"
```

**Review by Speuce — CHANGES_REQUESTED** [review]

File: frontend/src/components/modal/job/JobDetailModal.vue (hunk 2 above)
Comment (jaredwebber) [reply frontend/src/components/modal/job/JobDetailModal.vue:14]: This is like job title / job position not column/status (assuming thats what you were thinking) so should be string input

**Review by Speuce — APPROVED** [review]
LGTM

**Pull request merged**